### PR TITLE
Load data from API to summary fragment

### LINKED ISSUE #73: Load information from API to RepSummaryFragment
Let's just includes the things besides the issues and PR since those seems to be in work.

### PR DESCRIPTION
**Pull request by Huy-Ngo**
Draft to resolve #73

### CODE REVIEW

**Comment by McSinyx** [comment]
BTW if relevant enough please change the tab layout from fixed to scrollable.

File: app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+
+import io.noties.markwon.Markwon
+
+import vn.edu.usth.minigh.R
+import vn.edu.usth.minigh.api.github
+import vn.edu.usth.minigh.api.MinRepo
+
+class RepoSummaryFragment(val repo_name: String) : Fragment(R.layout.fragment_repo_summary) {
```
Comment (McSinyx) [inline app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt:25]: From [the docs](https://developer.android.com/reference/android/app/Fragment):

> All subclasses of Fragment must include a public no-argument constructor. The framework will often re-instantiate a fragment class when needed, in particular during state restore, and needs to be able to find this constructor to instantiate it. If the no-argument constructor is not available, a runtime exception will occur in some cases during state restore.

I suppose you want to use bundle to pass the arguments here (there's bundleOf FYI).

File: app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private var README: String = "Nothing to see here."
```
Comment (McSinyx) [inline app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt:26]: Is this variable really needed?

File: app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            }
+            starCount.text = "${repo.stargazers_count} Star" 
+            watchCount.text = "${repo.watchers_count} Watch"
+            forkCount.text = "${repo.forks_count} Fork"
+            readmeName.text = readme.name
+            if (readme.encoding == "base64") {
+                README = String(Base64.decode(readme.content, Base64.DEFAULT))
+                Markwon.create(getContext()!!).setMarkdown(readmeView, README)
+                val params = LinearLayout.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
+                readmeView.layoutParams = params
```
Comment (McSinyx) [inline app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt:61]: I suppose this should be applied to both cases of the conditional.

**Review by McSinyx — CHANGES_REQUESTED** [review]

File: app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt (hunk 2 above)
Comment (Huy-Ngo) [reply app/src/main/java/vn/edu/usth/minigh/fragments/RepoSummaryFragment.kt:26]: nope

**Review by McSinyx — APPROVED** [review]

**Pull request merged**